llvm/include/llvm/CodeGen/TargetLowering.h

```
@@ -2712,6 +2736,20 @@ class TargetLoweringBase {
       setCondCodeAction(CCs, VT, Action);
   }
 
+  /// Indicate how a PARTIAL_REDUCE_U/SMLA node with Acc type AccVT and Input
+  /// type InputVT should be treated by the target. Either it's legal, needs to
+  /// be promoted to a larger size, needs to be expanded to some other code
+  /// sequence, or the target has a custom expander for it.
+  void setPartialReduceMLAAction(MVT AccVT, MVT InputVT,
+                                 LegalizeAction Action) {
+    assert(AccVT.isValid() && InputVT.isValid() &&
```

nit: This could be a `nop` for `Action == Expand`

---

I'd rather not silently ignore calls to this, even if they could be a nop in some cases. 
In the (admittedly unlikely) case something tries to override an assigned action (think assigned to something in a loop, then reassigned to `Expand` following said loop), the second call would be ignored.

---

Ah yeah, that makes sense, ignore this comment :+1: